x/fundraising/keeper/invariants.go

```diff
@@ -0,0 +1,126 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/fundraising/x/fundraising/types"
+)
+
+// RegisterInvariants registers all fundraising invariants.
+func RegisterInvariants(ir sdk.InvariantRegistry, k Keeper) {
+	ir.RegisterRoute(types.ModuleName, "selling-pool-reserve-amount",
+		SellingPoolReserveAmountInvariant(k))
+	ir.RegisterRoute(types.ModuleName, "paying-pool-reserve-amount",
+		PayingPoolReserveAmountInvariant(k))
+	ir.RegisterRoute(types.ModuleName, "vesting-pool-reserve-amount",
+		VestingPoolReserveAmountInvariant(k))
+}
+
+// AllInvariants runs all invariants of the fundraising module.
+func AllInvariants(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		for _, inv := range []func(Keeper) sdk.Invariant{
+			SellingPoolReserveAmountInvariant,
+			PayingPoolReserveAmountInvariant,
+			VestingPoolReserveAmountInvariant,
+		} {
+			res, stop := inv(k)(ctx)
+			if stop {
+				return res, stop
+			}
+		}
+		return "", false
+	}
+}
+
+// SellingPoolReserveAmountInvariant checks an invariant that the total amount of selling coin for an auction
+// must equal to the selling reserve account balance.
+func SellingPoolReserveAmountInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		msg := ""
+		count := 0
+
+		for _, auction := range k.GetAuctions(ctx) {
+			if auction.GetStatus() == types.AuctionStatusStarted {
+				sellingPoolAcc := auction.GetSellingPoolAddress()
+				sellingReserve := k.bankKeeper.GetBalance(ctx, sellingPoolAcc, auction.GetSellingCoin().Denom)
+				if !sellingReserve.Equal(auction.GetSellingCoin()) {
```

Comment:
Since `sellingPoolAcc` is not a module account that blocks send, anyone can send assets here, and in this case, it can be larger than `auction.GetSellingCoin()`, so I think I can avoid panic by using `IsGTE` rather than `Equal`.